### ISSUE TITLE
Enable/test torch.stack for Q2 target models

### ISSUE THREAD

**Issue by michihirohorie** [torch-spyre/torch-spyre#1456]
### What feature would you like to be added?

Enable torch.stack

### Why is this feature needed?

torch.stack is used in Q2 target model: [Support Mistral 3 Small 24B](https://github.com/torch-spyre/torch-spyre/issues/698)

### Anything else we need to know?

#940

**Comment by michihirohorie** [comment]
Two issues exist.

### **Issue 1: torch.int64**
Input dtype is int64 in Mistral use

### **Issue 2: concatenating values that are smaller than the number of stick elements at stick dim** 

Fewer than 64 elements (in FP16) are concatenated along the stick dimension (dim=-1)


Test script (stacking two FP16 tensors with dim=-1), motivated by an actual parameter from [Mistral-Small-3.2-24B](https://github.com/torch-spyre/torch-spyre/blob/main/tests/resource/models/Mistral-Small-3.2-24B-Instruct-2506.yaml#L237-L263)
```
import torch


DEVICE = torch.device("spyre")
torch.manual_seed(0xAFFE)

x = torch.randn(28, 38, dtype=torch.float16)
y = torch.randn(28, 38, dtype=torch.float16)

cpu_result = torch.stack([x, y], dim=-1)

x_device = x.to(DEVICE)
y_device = y.to(DEVICE)

compiled = torch.compile(lambda a, b: torch.stack([a, b], dim=-1))
compiled_result = compiled(x_device, y_device).cpu()

# Print the results and compare them
print("Input\n", x, ", ", y)
print("CPU result\n", cpu_result)
print("Spyre Compiled result\n", compiled_result)

if not torch.equal(cpu_result, compiled_result):
    print(f"Not matched")
```

Torch.stack is decomposed with unsqueeze and cat (overwrite):

```
class <lambda>(torch.nn.Module):
    def forward(self, arg0_1: "f16[28, 38]", arg1_1: "f16[28, 38]"):
        # File: /home/horie/dt-inductor/torch-spyre/examples/stack.py:15 in <lambda>, code: compiled = torch.compile(lambda a, b: torch.stack([a, b], dim=-1))
        unsqueeze: "f16[28, 38, 1]" = torch.ops.aten.unsqueeze.default(arg0_1, 2);  arg0_1 = None
        unsqueeze_1: "f16[28, 38, 1]" = torch.ops.aten.unsqueeze.default(arg1_1, 2);  arg1_1 = None
        empty: "f16[28, 38, 2]" = torch.ops.aten.empty.memory_format([28, 38, 2], dtype = torch.float16, layout = torch.strided, device = device(type='spyre', index=0), pin_memory = False, memory_format = torch.contiguous_format)
        overwrite: "f16[28, 38, 2]" = torch.ops.spyre.overwrite.default(unsqueeze, empty, -1, 0);  unsqueeze = empty = None
        overwrite_1: "f16[28, 38, 2]" = torch.ops.spyre.overwrite.default(unsqueeze_1, overwrite, -1, 1);  unsqueeze_1 = overwrite = None
        return (overwrite_1,)
```

**Comment by ani300** [comment]
We also need torch.stack for some Rope issues with float16/float32

**Comment by moriohara** [comment]
toch.stack is used only in Mistral for Q2 target models, where torch.stack takes CPU tensors. That is we don't need to run it on Spyre.

https://github.com/torch-spyre/torch-spyre/blob/aed9854e7861899678af28fc0f883cfac965f6c8/tests/resource/models/Mistral-Small-3.2-24B-Instruct-2506_spyre.yaml#L221-L245
